backend/models/user.js

```diff
@@ -121,14 +157,18 @@ User.authenticate =  async function (logger, username, password) {
 			username = user.user;
 		}
 
-		authDB = await DB.getAuthDB();
-		await authDB.authenticate(username, password);
-		authDB.close();
-
 		if (!user)  {
 			user = await User.findByUserName(username);
 		}
 
+		if (isAccountLocked(user)) {
+			throw ({ resCode: responseCodes.TOO_MANY_LOGIN_ATTEMPTS });
+		}
+
+		authDB = await db.getAuthDB();
```

Comment:
can we refactor this to go go in to the dbhandler? 

so this file should only need to call `db.authenticate()`; and shouldn't need to deal with the db connections